app/models/base_facility.rb

```diff
@@ -201,6 +204,20 @@ def build_distance_result_set(lat, long, type, services, ids, limit = nil)
     end
     # rubocop:enable Metrics/ParameterLists
 
+    def zip_query(params)
+      # TODO: allow user to set distance from zip
+      zip_plus0 = params[:zip][0...5]
+      zcta = CSV.read(Rails.root.join('modules', 'va_facilities', '2018_Gaz_zcta_national.txt'), col_sep: "\t")
+      requested_zip = zcta.select { |area| area[0] == zip_plus0 }
+      requested_zip = [['x']] unless requested_zip.any?
+      # TODO: iterate over zcta, pushing each zip code that is within distance into an array
+      # TODO: change zip criteria to array of zip codes
+      conditions = "address ->'physical'->>'zip' ilike '#{requested_zip[0][0]}%'"
+      TYPES.map do |facility_type|
+        get_facility_data(conditions, params[:type], facility_type, params[:services])
+      end.flatten
+    end
```

Comment:
Probably going to extract this TYPES.map.flatten when we merge everything together, unless you want to do it preemptively. Any thoughts on what to call it? Maybe `get_for_all_facility_types`?

Reply:
`flat_map` can replace `map{...}.flatten`

Reply:
I don't love that name, but I've yet to come up with anything clearer. :man_shrugging: